ui/admin/src/app/libraries/library.ts

```diff
@@ -0,0 +1,35 @@
+export interface OpeningHours {
+  day: string;
+  is_open: boolean,
+  times: Array<Hours>;
+}
+
+export interface Hours {
+  start_time: string;
+  end_time?: string
+}
+
+export class Library {
+  $schema: string;
+  id: number;
+  pid: number;
+  name: string;
+  email: string;
+  address: string;
+  code: string;
+  opening_hours: Array<OpeningHours>;
+
+  constructor(obj?: any) {
+    Object.assign(this, obj);
+    this.opening_hours = this.populateTimes(this.opening_hours);
+  }
+
+  populateTimes(openinghours) {
+    for (let hours in openinghours) {
```

Comment:
it is probably better to use` for..of` instead of `for...in`: https://basarat.gitbooks.io/typescript/docs/for...of.html
or even better: `openinghours.map()`:
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/map

Reply:
In this code, i use key and not value.